src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AlleleFractionModeller.java

```diff
@@ -103,25 +102,29 @@ public Double sample(final RandomGenerator rng, final AlleleFractionState state,
         }
     }
 
+    private static Function<Double, Double> logConditionalOnMinorFraction(final AlleleFractionState state,
+            final AlleleFractionData data, final int segment) {
+        return minorFraction -> {
+            final AlleleFractionState proposal = makeSingleSegmentState(state.meanBias(),
+                    state.biasVariance(), state.outlierProbability(), minorFraction);
+            return segmentLogLikelihood(proposal, 0, data.countsInSegment(segment));
+        };
+    }
+
     // sample minor fraction of a single segment
     private static final class PerSegmentMinorFractionSampler implements Sampler<Double, AlleleFractionState, AlleleFractionData> {
-        private static final double INITIAL_MINOR_FRACTION_STEP_SIZE = 0.05;
         private static final double MAX_MINOR_FRACTION = 0.5;   //by definition!
         private final AdaptiveMetropolisSampler sampler;
 
         private final int segmentIndex;
 
-        public PerSegmentMinorFractionSampler(final int segmentIndex, final double initialMinorFraction) {
-            sampler = new AdaptiveMetropolisSampler(initialMinorFraction, INITIAL_MINOR_FRACTION_STEP_SIZE, 0, MAX_MINOR_FRACTION);
+        public PerSegmentMinorFractionSampler(final int segmentIndex, final double initialMinorFraction, final double initialStepSize) {
+            sampler = new AdaptiveMetropolisSampler(initialMinorFraction, initialStepSize, 0, MAX_MINOR_FRACTION);
             this.segmentIndex = segmentIndex;
         }
 
         public Double sample(final RandomGenerator rng, final AlleleFractionState state, final AlleleFractionData data) {
```

Comment:
note when rebasing that i added the following here to handle missing data:

    if (data.numHetsInSegment(segmentIndex) == 0) {
        return Double.NaN;
    }